projects/igniteui-angular/src/lib/core/displayDensity.ts

```diff
@@ -55,20 +62,31 @@ export class DisplayDensityBase {
             default:
                 this._displayDensity = DisplayDensity.comfortable;
         }
+        if (currentDisplayDensity !== this._displayDensity) {
+            const densityChangedArgs: IDensityChangedEventArgs = {
+                oldDensity: currentDisplayDensity,
+                newDensity: this._displayDensity
+            };
+        this.onDensityChanged.emit(densityChangedArgs);
+        }
     }
 
+    @Output()
+    public onDensityChanged = new EventEmitter<IDensityChangedEventArgs>();
+    protected oldDisplayDensityOptions: IDisplayDensity = { displayDensity: DisplayDensity.comfortable };
```

Comment:
Do we need both `protected _displayDensity: DisplayDensity | string;` and `protected oldDisplayDensityOptions: IDisplayDensity = { displayDensity: DisplayDensity.comfortable };`? These two members are holding the exact same information and needs to be in sync.

Reply:
Also `protected displayDensityOptions: IDisplayDensity`.

Reply:
_displayDensity is the value set for the control`s displayDensity input and oldDisplayDensityOptions holds the previous value of the global display density. This mechanism is needed in order to detect changes in the global displayDenistyOptions and trigger event only then for the grid in order to update its values

Reply:
Ok. Thanks.